lib/galaxy/datatypes/proteomics.py

```diff
@@ -0,0 +1,405 @@
+"""
+Proteomics Datatypes
+"""
+import logging
+import re
+import binascii
+
+from galaxy.datatypes import data
+from galaxy.datatypes.data import Text
+from galaxy.datatypes.xml import GenericXml
+from galaxy.datatypes.binary import Binary
+from galaxy.datatypes.tabular import Tabular
+
+log = logging.getLogger(__name__)
+
+
+class Wiff(Binary):
+    """Class for wiff files."""
+    file_ext = 'wiff'
+    allow_datatype_change = False
+    composite_type = 'auto_primary_file'
+
+    def __init__(self, **kwd):
+        Binary.__init__(self, **kwd)
+
+        self.add_composite_file(
+            'wiff',
+            description='AB SCIEX files in .wiff format. This can contain all needed information or only metadata.',
+            is_binary=True)
+
+        self.add_composite_file(
+            'wiff_scan',
+            description='AB SCIEX spectra file (wiff.scan), if the corresponding .wiff file only contains metadata.',
+            optional='True', is_binary=True)
+
+    def generate_primary_file(self, dataset=None):
+        rval = ['<html><head><title>Wiff Composite Dataset </title></head><p/>']
+        rval.append('<div>This composite dataset is composed of the following files:<p/><ul>')
+        for composite_name, composite_file in self.get_composite_files(dataset=dataset).iteritems():
+            fn = composite_name
+            opt_text = ''
+            if composite_file.optional:
+                opt_text = ' (optional)'
+            if composite_file.get('description'):
+                rval.append('<li><a href="%s" type="text/plain">%s (%s)</a>%s</li>' % (fn, fn, composite_file.get('description'), opt_text))
+            else:
+                rval.append('<li><a href="%s" type="text/plain">%s</a>%s</li>' % (fn, fn, opt_text))
+        rval.append('</ul></div></html>')
+        return "\n".join(rval)
+
+if hasattr(Binary, 'register_unsniffable_binary_ext'):
+    Binary.register_unsniffable_binary_ext('wiff')
```

Comment:
It doesn't hurt anything technically - but these ``hasattr`` checks are no longer needed (especially for code in core with the actual base class). So can you remove these checks - so people don't copy and paste the needed code for new binary types.